src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,1053 @@
+                        DoCast(SPELL_SUMMON_COLDWEAVE);
+                        DoCast(SPELL_SUMMON_COLDWEAVE);
+                        _events.ScheduleEvent(EVENT_SUMMON_COLDWEAVE, 8000, 0, PHASE_ONE);
+                        if (_submerged)
+                            _events.ScheduleEvent(EVENT_SUMMON_FROSTWIND, 4000, 0, PHASE_ONE);
+                        break;
+                    case EVENT_SUMMON_FROSTWIND:
+                        DoCast(SPELL_SUMMON_FROSTWIND);
+                        break;
+                    case EVENT_SUMMON_AHUNE:
+                        if (TempSummon* ahune = me->SummonCreature(NPC_AHUNE, SummonPositions[0], TEMPSUMMON_DEAD_DESPAWN))
+                        {
+                            ahune->SummonCreature(NPC_FROZEN_CORE, SummonPositions[1], TEMPSUMMON_CORPSE_DESPAWN);
+                            ahune->SetInCombatWithZone();
+                            DoCast(ahune, SPELL_RESURFACE);
+                        }
+                        break;
+                    case EVENT_CLOSE_OPENING:
+                        if (Creature* flamecaller = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_FLAMECALLER_000)))
+                            flamecaller->AI()->DoAction(ACTION_EMOTE_RESURFACE);
+                        DoCast(SPELL_CLOSE_OPENING_VISUAL);
+                        DoCast(me, SPELL_ICE_BOMBARD);
+                        break;
+                    case EVENT_AHUNE_PHASE_TWO:
+                        if (Creature* flamecaller = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_FLAMECALLER_000)))
+                            DoCast(flamecaller, SPELL_FOUND_OPENING);
+                        if (Creature* ahune = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_AHUNE)))
+                            ahune->AI()->DoAction(ACTION_AHUNE_RETREAT);
+                        _events.Reset();
+                        _events.SetPhase(PHASE_TWO);
+                        _events.ScheduleEvent(EVENT_CLOSE_OPENING, 25000, 0, PHASE_TWO);
+                        _events.ScheduleEvent(EVENT_AHUNE_PHASE_ONE, 35000, 0, PHASE_TWO);
+                        break;
+                    case EVENT_AHUNE_PHASE_ONE:
+                        _submerged = true;
+                        _events.Reset();
+                        _events.SetPhase(PHASE_ONE);
+                        _events.ScheduleEvent(EVENT_SUMMON_COLDWEAVE, 8000, 0, PHASE_ONE);
+                        _events.ScheduleEvent(EVENT_SUMMON_HAILSTONE, 5000, 0, PHASE_ONE);
+                        _events.ScheduleEvent(EVENT_START_LOOKING_FOR_OPENING, 5000, 0, PHASE_ONE);
+                        _events.ScheduleEvent(EVENT_AHUNE_PHASE_TWO, 100000, 0, PHASE_ONE);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        void ResetFlameCallers()
+        {
+            for (int counter = 0; counter < MAX_FLAMECALLERS; ++counter)
```

Comment:
The comment by vm is bad (using int isnt "c style") but he is correct that there is an issue here - we do not use plain "int" type - use uint32